framework/src/org/checkerframework/framework/type/SupertypeFinder.java

```diff
@@ -189,6 +196,51 @@
             return supertypes;
         }
 
+        /**
+         * If the upper bound of the wildcard is not a subtype of the upper bound of the type
+         * parameter, then this method fixes that.
+         *
+         * <p>This method is needed because, the Java compiler allows wildcards to have upper bounds
+         * above the type variable upper bounds for which they are type arguments. For example,
+         * given the following parametric type:
+         *
+         * <pre>
+         * {@code class MyClass<T extends Number>}
+         * </pre>
+         *
+         * the following is legal:
+         *
+         * <pre>
+         * {@code MyClass<? extends Object>}
+         * </pre>
+         *
+         * This is sound because in Java the wildcard is capture converted to: {@code CAP#1 extends
+         * Number from capture of ? extends Object}.
+         *
+         * <p>Because the Checker Framework does not implement capture conversion, the wildcard's
+         * upper bound may not be within the type parameter. This method fixes that.
+         *
+         * @param element type parameter to which {@code wildcard} is an argument
+         * @param wildcard wildcard type whose upper bound may be modified.
+         * @return
+         */
+        private void fixWildcardBound(
```

Comment:
We could additionally store "original" super/extends bounds with the AnnotatedWildcardType and use that in error messages.
I think it would be better to do this "widenToUpperbound" once in the AWT and make sure we never run into this same issue again.